### PR TITLE
[WIP] Riiif

### PR DESCRIPTION
**Pull request by hackmastera**
see #617 

Here for discussion on commits so far, as desired.

### CODE REVIEW

File: app/assets/javascripts/chf_image_viewer.js [hunk 1]
```diff
@@ -124,10 +117,11 @@ ChfImageViewer.prototype.selectThumb = function(thumbElement) {
   var shouldShowInfo = thumbElement.getAttribute('data-member-should-show-info') == "true";
   var title = thumbElement.getAttribute('data-title');
   var linkUrl   = thumbElement.getAttribute('data-member-show-url');
+  var titleSource = thumbElement.getAttribute('data-title-source');
 
   $('.viewer-image').addClass('viewer-image-loading');
 
-  this.viewer.open(this.id2TileSource(id));
+  this.viewer.open(titleSource);
```
Comment (jrochkind) [inline app/assets/javascripts/chf_image_viewer.js]: what's "title" about this?  A typo for `tile-source`?

File: app/helpers/riiif_helper.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module RiiifHelper
+  # use relative url unless we've defind a riiif server in config/environments/*.rb
+  def riiif_image_url (file_set_id)
+    file_id = FileSet.find(file_set_id).original_file.id
+    path = riiif.info_path(file_id, locale: nil)
+    begin
+      return URI.join(Rails.configuration.riiif_server, path).to_s
+    rescue NoMethodError
+      return path
```
Comment (jrochkind) [inline app/helpers/riiif_helper.rb]: what circumstances is this rescuing from, what method ends up missing?  If you don't have `Rails.configuration.riiif_server`?  I think a `Rails.configuration.respond_to?(:riiif_server)` is better than a rescue.

File: app/assets/javascripts/chf_image_viewer.js (hunk 1 above)
Comment (hackmastera) [inline app/assets/javascripts/chf_image_viewer.js]: i think my wires were crossed a bit. 'tile source' makes so much more sense; I kept wondering why it was called 'title source'.

File: app/helpers/riiif_helper.rb (hunk 2 above)
Comment (hackmastera) [inline app/helpers/riiif_helper.rb]: sounds good.

**Comment by hackmastera** [comment]
@jrochkind thanks for the review so far; i believe I've addressed all your points up to now.

**Comment by jrochkind** [comment]
Is this ready for merge, as far as riiif functionality goes?

Before we actually merge to master, we need to deal with purging cached files -- whether it's going to be on the app server or a riiif server with more expansive disk space will effect purge strategies. Either way, could be a cronjob deployed by cap, or something in ansible. But we shouldn't deploy to prod until we have _something_ there, so shouldn't merge to master either, I think?

But we could play with this on staging. And I might work on top of it to do other things I was on waiting for riiif. eg #635 #636

**Comment by hackmastera** [comment]
This PR is against master, but you could work against this branch if you like.

Outstanding TODOs are on https://github.com/chemheritage/chf-sufia/issues/617

All the commits here have been reviewed. would you like me to submit the next one as a PR against this branch?

**Comment by hackmastera** [comment]
But yes, riiif stuff works on this branch you could do use this as a starting point for those other issues you mentioned.

**Comment by jrochkind** [comment]
Ah, I see, so you don't plan to merge this to master until all the checkboxes on #617 are marked?

I don't have clarity myself on whether those should be PR's against this or master, what was your original plan or what do you think would be most convenient?

**Comment by hackmastera** [comment]
@jrochkind right, i wasn't going to merge to master until that ticket is done. Just like you said above: "Before we actually merge to master, we need to deal with purging cached files -- whether it's going to be on the app server or a riiif server with more expansive disk space will effect purge strategies"...

i'm working on the indexing piece right now; nearly done. the caching stuff will be all that's left after that.

I think it will be easier for you to review if I PR against this branch, since you've already reviewed what's here.

File: config/initializers/riiif.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  fs_id = id.sub(/\A([^\/]*)\/.*/, '\1')
+  resp = ActiveFedora::SolrService.get("id:#{fs_id}")
+  doc = resp['response']['docs'].first
+  raise "Unable to find solr document with id:#{fs_id}" unless doc
+  { height: doc['height_is'], width: doc['width_is'] }
+end
+
+def logger
+  Rails.logger
+end
```
Comment (jrochkind) [inline config/initializers/riiif.rb]: Missed this before, but this isn't right. You are effectively defining a logger method at top-level ruby scope. Or somewhere else that probably isn't what we want. 

What was intended here? Did this actually come from a riiif-generated template? grrr.

Comment (hackmastera) [inline config/initializers/riiif.rb]: oops, sorry. I pulled it from hyku as per https://github.com/RepoCamp/ahc/wiki/IIIF#install-a-riiif-config

here's the file I used: https://github.com/samvera-labs/hyku/blob/master/config/initializers/riiif.rb

Comment (jrochkind) [inline config/initializers/riiif.rb]: I dunno, I think hyku is very wrong there, don't understand what it's doing.

**Pull request merged**